profiles/lib/tag_cache.lua

```diff
@@ -0,0 +1,35 @@
+TagCache = {}
+
+function TagCache.get(data, key)
+  local v = data.cache[key]
+  if v then
+    if v == '' then
```

Comment:
Maybe we can move this mechanism to C++ land. I would like to remove this class because it works around broken binding behavior.

@daniel-j-h @MoKob any opinions here?